tests/skills/github/test_wait_for_unresolved_zero.py

```
@@ -0,0 +1,640 @@
+        self.assertEqual((count, complete, exit_code), (-1, False, 0))
+
+    def test_non_dict_payload_returns_negative(self) -> None:
+        """NEGATIVE: JSON is valid but not a dict (null/list/scalar) -> count=-1.
+
+        PR #1989 coderabbit njY: json.loads can return non-dict types;
+        calling .get() on them raises AttributeError. Defensive isinstance
+        check returns the sentinel cleanly.
+        """
+        for payload_str in ("null", "[]", '"a string"', "42"):
+            def runner(*_args: Any, **_kw: Any) -> Any:
+                return mock.Mock(returncode=0, stdout=payload_str, stderr="")
```

Ruff (0.15.12)

[warning] Function definition does not bind loop variable `payload_str`

(B023)